utils/src/offset/copy.rs

```diff
@@ -0,0 +1,264 @@
+use std::{mem, ptr};
+
+use light_bounded_vec::{
+    BoundedVec, BoundedVecMetadata, CyclicBoundedVec, CyclicBoundedVecMetadata,
+};
+
+/// Creates a copy of value of type `T` based on the provided `bytes` buffer.
+///
+/// # Safety
+///
+/// This is higly unsafe. This function doesn't ensure alignment and
+/// correctness of provided buffer. The responsibility of such checks is on
+/// the caller.
+pub unsafe fn read_value_at<T>(bytes: &[u8], offset: &mut usize) -> T
+where
+    T: Clone,
+{
+    let size = mem::size_of::<T>();
+    let ptr = bytes[*offset..*offset + size].as_ptr() as *const T;
+    *offset += size;
+    // (*ptr).clone()
+    ptr::read(ptr)
+}
+
+/// Creates a `BoundedVec` from the sequence of values provided in `bytes` buffer.
+///
+/// # Safety
+///
+/// This is higly unsafe. This function doesn't ensure alignment and
+/// correctness of provided buffer. The responsibility of such checks is on
+/// the caller.
+pub unsafe fn read_bounded_vec_at<T>(
+    bytes: &[u8],
+    offset: &mut usize,
+    metadata: &BoundedVecMetadata,
+) -> BoundedVec<T>
+where
+    T: Clone,
+{
+    let size = mem::size_of::<T>() * metadata.capacity();
```

Comment:
We could add a safety comment that you cannot use this if T has properties that are vectors.

Reply:
Done, included it in the docstring